pkg/kv/mem/store.go

```diff
@@ -39,23 +41,45 @@ func (d *Driver) Open(_ context.Context, _ string) (kv.Store, error) {
 	}, nil
 }
 
-func (s *Store) Get(_ context.Context, key []byte) (*kv.ValueWithPredicate, error) {
+func combinedKey(partitionKey, key []byte) string {
+	return fmt.Sprintf("%s_%s", partitionKey, key)
+}
+
+func keyFromCombinedKey(combinedKey string) []byte {
+	//nolint:gomnd
+	return []byte(strings.SplitN(combinedKey, "_", 2)[1])
+}
+
+func partitionKeyFromCombinedKey(combinedKey string) []byte {
+	//nolint:gomnd
+	return []byte(strings.SplitN(combinedKey, "_", 2)[0])
+}
```

Comment:
I believe the usage of `'_'` as a separator can be easily broken, as nothing prevent a [artitionKey to contain `'_'`.
I know that `DynamoDB` doesn't prevent the use of `'_'` as part of ket, partition or value, so not sure what should be the correct behavior here.
Nevertheless, I don't think it is a blocker for this PR, certainley not at that point, but we might want to address it in a followup issue.